sdk/azidentity/default_azure_credential.go

```diff
@@ -158,7 +158,7 @@ type defaultCredentialErrorReporter struct {
 }
 
 func (d *defaultCredentialErrorReporter) GetToken(ctx context.Context, opts policy.TokenRequestOptions) (azcore.AccessToken, error) {
-	if _, ok := d.err.(*credentialUnavailableError); ok {
+	if _, ok := d.err.(credentialUnavailable); ok {
```

Comment:
Hi @chlowell! Is there a reason to leave the credentialUnavailable type unexported? Trying to implement a credentialErrorReporter wrapper for our custom ChainedTokenCredential but can't due to this line. Thank you!

Reply:
We didn't see a need to export the interface, and we generally prefer to keep public API surfaces as small as possible. What do you need `credentialUnavailable` for in your code? `defaultCredentialErrorReporter` doesn't need this line to function correctly.

Reply:
> defaultCredentialErrorReporter doesn't need this line to function correctly.
Got it, had figured I could just unconditionally new it but the presence of the additional type assertion check had me questioning myself. I guess it's in the defaultCredentialErrorReporter just to save unnecessary new-s. Thanks!